src/SonarScanner.MSBuild.PreProcessor/ProcessedArgs.cs

```diff
@@ -286,6 +295,45 @@ ServerInfo Warn(ServerInfo server, string message)
                 return server;
             }
         }
+
+        private bool TryGetUserHome(ILogger logger, IDirectoryWrapper directoryWrapper, IOperatingSystemProvider operatingSystemProvider, out string userHome)
+        {
+            if (AggregateProperties.TryGetProperty(SonarProperties.UserHome, out var userHomeProp))
+            {
+                if (directoryWrapper.Exists(userHomeProp.Value))
+                {
+                    userHome = userHomeProp.Value;
+                    return true;
+                }
+                else
+                {
+                    logger.LogError(Resources.ERR_UserHomeInvalid, userHomeProp.Value);
+                    userHome = null;
+                    return false;
+                }
+            }
+            var defaultPath = Path.Combine(operatingSystemProvider.GetFolderPath(Environment.SpecialFolder.UserProfile, Environment.SpecialFolderOption.None), ".sonar");
+            if (directoryWrapper.Exists(defaultPath))
+            {
+                userHome = defaultPath;
+                return true;
+            }
+            else
+            {
+                try
+                {
+                    directoryWrapper.CreateDirectory(defaultPath);
+                    userHome = defaultPath;
+                    return true;
+                }
+                catch (Exception ex)
+                {
+                    logger.LogWarning(Resources.WARN_DefaultUserHomeCreationFailed, defaultPath, ex.Message);
+                    userHome = null;
+                    return false;
+                }
+            }
```

Comment:
Can be simplified a bit
```suggestion
            if (!directoryWrapper.Exists(defaultPath))
            {
                try
                {
                    directoryWrapper.CreateDirectory(defaultPath);
                }
                catch (Exception ex)
                {
                    logger.LogWarning(Resources.WARN_DefaultUserHomeCreationFailed, defaultPath, ex.Message);
                    userHome = null;
                    return false;
                }
            }
            userHome = defaultPath;
            return true;
```

Reply:
`return false;` is actually wrong here: We either log an error and return false or we log a warning and return true. I intended to do the latter: We only need the userHome directory for the JRE caching. JRE caching is not supposed to succeed, so the userHome is not mandatory, and we should move on instead of failing.
What do you think?

Reply:
We do have these outcomes now
| | default | sonar.userHome |
| --- | --- | --- |
| Create successful | successful, returns path, no messages |  successful, returns path, debug message about created directory |
| Create failed | successful, returns null, warning message | failed, returns null, error message |

We need the Try pattern because null can mean "Success and failure" depending on whether sonar.userHome is set or not.